core/PaginatedList.php

```diff
@@ -229,9 +229,19 @@ public function Pages($max = null) {
 		}
 
 		for ($i = $start; $i < $end; $i++) {
+
+			// Remove the query sting for the first page link
+			// /?start=0 returns the same page as /
+			$link = HTTP::setGetVar($this->getPaginationGetVar(), $i * $this->getPageLength());
+
+			if($i === 0) {
+				$parts = explode("?", $link);
```

Comment:
I suggest parsing out querystring args, removing the `start` argument, then re-encoding back into a string.